en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/troubleshooting/debug-memory.md

```diff
@@ -0,0 +1,39 @@
+# Debugging Memory Issues
+
+As with any software product, there can be instances where WSO2 Choreo Connect cluster fails due to a memory resource exhaustion. The heap dumps will always point you towards the cause of the memory leak. Therefore, it is important to be able to retrieve heap dumps from an environment at the point when an error occurs. This will avoid the necessity of reproducing the exact issue again (particularly in the case of production issues). A resource exhaustion can happen for two reasons:
+
+- Due to a bug in the system.
+- An actual limitation of resources based on low configuration values.
+
+Following sections will guide you on how to take heap dumps from each component of WSO2 Choreo Connect.
+
+## Taking a heap dump of Adapter
+
+The Adapter is implemented in Go lang and, therefore, it exposes standard `pprof` ([GO pprof pkg](https://pkg.go.dev/net/http/pprof)) based Go profiling endpoints in `localhost`.
+Follow the steps below to take a heap dump of the Adapter.
+
+1. Log into the Adapter container's shell or in kubernetes, port-forward the port `6060` from Adapter to localhost. This step is required because Adapter's `pprof` endpoints are only exposed via localhost for safety.
```

Comment:
```suggestion
1. Log into the Adapter container's shell or in Kubernetes, port-forward the port `6060` from Adapter to localhost. This step is required because Adapter's `pprof` endpoints are only exposed via localhost for safety.
```